sessionctx/variable/sysvar.go

```diff
@@ -683,6 +683,30 @@ var defaultSysVars = []*SysVar{
 			return nil
 		},
 	},
+	{Scope: ScopeGlobal, Name: TiDBEnablePreparedPlanCache, Value: BoolToOnOff(DefTiDBEnablePreparedPlanCache), Type: TypeBool, SetGlobal: func(s *SessionVars, val string) error {
+		EnablePreparedPlanCache.Store(TiDBOptOn(val))
+		return nil
+	}, GetGlobal: func(s *SessionVars) (string, error) {
+		return BoolToOnOff(EnablePreparedPlanCache.Load()), nil
+	}},
+	{Scope: ScopeGlobal, Name: TiDBPreparedPlanCacheSize, Value: strconv.FormatUint(uint64(DefTiDBPreparedPlanCacheSize), 10), Type: TypeUnsigned, MinValue: 1, MaxValue: 100000, SetGlobal: func(s *SessionVars, val string) error {
+		uVal, err := strconv.ParseUint(val, 10, 64)
+		if err == nil {
+			PreparedPlanCacheSize.Store(uint32(uVal))
+		}
+		return err
+	}, GetGlobal: func(s *SessionVars) (string, error) {
+		return strconv.FormatUint(uint64(PreparedPlanCacheSize.Load()), 10), nil
+	}},
```

Comment:
I have a suggestion actually. We should be able to merge `TiDBPreparedPlanCacheSize` and `TiDBEnablePreparedPlanCache` into one sysvar:

* TiDBPreparedPlanCacheSize`

We can then add logic so that if the size is > 0 it is enabled. The default will have to be changed from `1000` to `0` to respect the previous behavior.